syntax = "proto3";
option csharp_namespace = "inference_server";

package inference;

// Inference service definition
service Inference {
  // Runs a model inference
  rpc RunInference (InferenceParameters) returns (InferenceResult);
}

// Inference input parameters
message InferenceParameters {
  string modelHash = 1;
  string modelInput = 2;
}

// The response message containing the inference result
message InferenceResult {
  string value = 1;
}
